showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -127,6 +155,93 @@ class ShowkaseProcessor @JvmOverloads constructor(
+                }
+                showkaseValidator.validateComponentElementOrSkip(
+                    element = annotatedElement,
+                    annotationName = supportedType,
+                )
+                components.addAll(
+                    getShowkaseMetadataFromCustomAnnotation(
+                        element = annotatedElement,
+                        showkaseValidator = showkaseValidator,
+                        supportedType.getCustomAnnotationSimpleName(),
+                    ).toSet()
+                )
+            }
+        }
+        return components
+    }
+
+    private fun String.getCustomAnnotationSimpleName(): String {
+        return this.split(".").last()
+    }
+
+    private fun processCustomAnnotationFromClasspath(roundEnvironment: XRoundEnv): Set<ShowkaseMetadata.Component> {
+        // In this function we are checking generated classpath for MultiPreview codegen annotations.
+        // We also check the current module if there is any composables that are annotated with the qualified name
+        // from the annotation from classpath. We use the fields from the classpath annotation to build
+        // common data for the ShowkasMetadata.
+
+        // Supported annotations from classpath
+        val supportedCustomPreview = environment.getTypeElementsFromPackage(CODEGEN_PACKAGE_NAME)
+            .flatMap { it.getEnclosedElements() }.mapNotNull {
+                when (
+                    val annotation = it.getAnnotation(ShowkaseMultiPreviewCodegenMetadata::class)
+                ) {
+                    null -> null
+                    else -> annotation.value
+                }
+            }
+        val components = mutableSetOf<ShowkaseMetadata.Component>()
+
+        supportedCustomPreview
+            .mapIndexed { index: Int, customPreviewMetadata: ShowkaseMultiPreviewCodegenMetadata ->
+                roundEnvironment
+                    .getElementsAnnotatedWith(customPreviewMetadata.supportTypeQualifiedName)
+                    .mapIndexed elementRoot@{ elementIndex, xElement ->
+
+                        if (showkaseValidator.checkElementIsAnnotationClass(xElement)) {
+                            // Here we write to metadata to aggregate custom annotation data.
+                            // In this case, it would be a custom annotation that is annotated
+                            // with something custom. Eg. CombinedPreview.
+                            ShowkaseBrowserWriter(environment).writeCustomAnnotationElementToMetadata(
+                                xElement
+                            )
+                            return@elementRoot null
+                        }
+                        showkaseValidator.validateComponentElementOrSkip(
+                            xElement,
+                            customPreviewMetadata.supportTypeQualifiedName,
+                        )
+                        components.add(
```

Comment:
you don't need to add it to a set like this. Just collect the whole thing in a variable. For example

```kotlin
return supportedCustomPreview
            .mapIndexed {...}.toSet()
```

Reply:
I think we can do it like that, it was just that since it had two mapIndexed method, I found this was a bit easier to read. I can ofc change it if you think your way is easier :) Wdyt? :)